symbol-overlay.el

```diff
@@ -235,21 +235,21 @@ You can re-bind the commands to any keys you prefer.")
   "Get all highlighted overlays in the buffer.
 If SYMBOL is non-nil, get the overlays that belong to it.
 DIR is an integer.
+  If DIR = 0, search the whole buffer;
+  If DIR > 0, search the region after current point;
+  If DIR < 0, search the region before current point.
 If EXCLUDE is non-nil, get all overlays excluding those belong to SYMBOL."
-  (if (= dir 0)
-      (overlays-in (point-min) (point-max))
-    (let ((overlays (cond ((< dir 0) (overlays-in (point-min) (point)))
-                          ((> dir 0) (overlays-in (point) (point-max))))))
-      (seq-filter
-       (lambda (ov)
-         (let ((value (overlay-get ov 'symbol))
-               (end (overlay-end ov)))
-           (and value
-                (or (> dir 0) (< end (point)))
```

Comment:
I don't think this check should be removed; otherwise, `symbol-overlay-get-lists` will return an incorrect list of overlays:

<img width="164" alt="image" src="https://user-images.githubusercontent.com/23358293/208439791-76aaacf4-1549-4ecd-a0ec-42524794d5ef.png">

Reply:
It seems that the problem is caused by the function "symbol-overlay-maybe-count", of which the counting is wrong.
I think the logic in "symbol-overlay-get-list" is correct, it correctly returns all the overlays.
I add a new commit to fix "symbol-overlay-maybe-count".

Reply:
Please just ignore my previous comment.
If I think the logic in "symbol-overlay-get-list" is correct, it needs to adapt more other functions which uses "symbol-overlay-get-list", it is not good idea.
So that I re-add the removed lines in "symbol-overlay-get-list".